### PR TITLE
User details update

### PR DESCRIPTION
**Pull request by burakkarasel**
Updated Role Enum for the needs. 
Changed User Entity to store only auth related fields. 
Changed Entity ID's from Integer to UUID. 
Added new Gender Enum.
Added new UserOtherDetails Entity to store non-auth related fields.
Created a OneToOne relation between User and UserOtherDetails.
Created new Service and Repository Implementations for UserOtherDetails.
Updated register functionality for the new changes.
Added new dependency for input validation.
Added annotations to request dtos.
Added custom exception handler for input validation.
Added new dependency ObjectMapper to convert test bodies to json.
Added new unit tests for register and authenticate functionality.
Updated regex pattern for email and password validation.
Added new AuthenticationExceptionHandler.
Added new UserService.
Added new InsuranceInformationService.
Added new RoleChangeNotPossibleException and it's handler.
Added new CreatePatientRequest and CreateResponse dtos.
Implemented Create a new Patient functionality.
Added new SpringSecurityUserProvider test configuration to mock authenticated user.
Changed AuthenticationController's location from service to controller package.
Added new PatientControllerTest to test creating a new patient functionality.
Changed Role enum, added PatientDto.
Added new repository and service method to find patient by user.
Added controller for patient details with pre authorization for patient role.
Added new test cases for the new service

### CODE REVIEW

**Comment by mustafaboleken** [comment]
Other than my notes, I will extend the unit tests. Then, all of us can use the same aproach.

**Comment by mustafaboleken** [comment]
We need to add logs

**Comment by mustafaboleken** [comment]
I just integrate the open api 3, you can pull from main and add API descriptions

**Comment by mustafaboleken** [comment]
Existing tests are not passing

**Comment by burakkarasel** [comment]
Added new NoSuchElementExceptionHandler.
Updated patient create service to accept UserDetails not User entity.
Updated tests accordingly.
Corrected few misspellings

**Comment by burakkarasel** [comment]
Added new unique restriction to User entity.
Added new DataIntegrityViolationExceptionHandler to handle the exceptions.
Added new test case for register with duplicate email.

**Review by mustafaboleken — APPROVED** [review]
lgtm

File: docker-compose.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ services:
     environment:
       - POSTGRES_USER=postgres
       - POSTGRES_PASSWORD=postgrespw
+      - POSTGRES_DATABASE=heu_plus
```
[the remark this answers is not in the document]
Comment (burakkarasel) [reply docker-compose.yml:9]: Not really😄

File: src/main/java/com/bola/boilerplate/service/concretes/PatientService.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                .build();
+
+        repository.save(toCreate);
+
+        return new CreateResponse("Patient created successfully");
+    }
+
+    @Override
+    public PatientDto details(User user) {
+        Patient patient = repository.findByUser(user);
```
[the remark this answers is not in the document]
Comment (burakkarasel) [reply src/main/java/com/bola/boilerplate/service/concretes/PatientService.java]: In the updated version I've used orElseThrow() which throws NoSuchElementException and it's handled by it's custom exception handler

File: src/main/java/com/bola/boilerplate/service/concretes/InsuranceInformationService.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+public class InsuranceInformationService implements InsuranceInformationManager {
+    private final InsuranceInformationRepository repository;
+    @Override
+    public InsuranceInformation create(String providerName, String policyNumber) {
+        InsuranceInformation toCreate = InsuranceInformation.builder()
+                .providerName(providerName)
+                .policyNumber(policyNumber)
+                .build();
+
+        return repository.save(toCreate);
```
[the remark this answers is not in the document]
Comment (burakkarasel) [reply src/main/java/com/bola/boilerplate/service/concretes/InsuranceInformationService.java]: Handling now #6

**Pull request merged**